docs/data-model.rst

```diff
@@ -997,9 +997,22 @@ Consider the following example:
 
 In this tree, node 4 is isolated, and therefore for any sites that are
 on this tree, the state that it is assigned is a special value
-``tskit.MISSING_DATA``, or ``-1``. See the :meth:`TreeSequence.variants`
-method and :class:`Variant` class for more information on how missing
-data is represented.
+``tskit.MISSING_DATA``, or ``-1``. Isolated nodes in a tree can be identified
+using the :meth:`Tree.is_isolated` method. Note that, although isolated, node
+4 above is still considered as being present in the tree. This means, for example,
+that the node is still returned by the :meth:`Tree.nodes` and :meth:`Tree.samples`
+methods. We can therefore identify isolated nodes, as well as nodes linked to the
+main tree topology, when iterating using these methods:
+
+    >>> [u for u in tree.nodes() if tree.is_isolated(u)]       # list isolated_nodes
+    [4]
+    >>> [u for u in tree.nodes() if not tree.is_isolated(u)]   # list nodes in topology
```

Comment:
This is probably redundant, I think? These are non-isolated nodes, so pretty obvious, and avoids semantic issues with the description in your comment (see previous comment).

Reply:
As you mentioned in https://github.com/tskit-dev/tskit/issues/442#issuecomment-562221369, it's possibly more useful to list the non-isolated nodes than the isolated ones. That's why I left this example in, really, even though it's a bit redundant. I could change he comment to simply "list non-isolated nodes" (although that makes it sound *really* redundant!)

Reply:
I still think it is useful to point out that if someone wants to print out the topologically connected nodes, they can use the `is_isolated` method. It's not going to be obvious to most people who deal with trees that, when looking at trees in a tree sequence, there will be nodes  which are not connected to the topology, but instead separate (isolated). Unless you know already about how missing data is encoded, this must come as a bit of a surprise. It certainly fooled me: I expected to be able to iterate over the nodes in a tree topology by doing `tree.nodes()`, and had to dig a bit to realise I had to actively exclude the "missing" ones.

Perhaps I'm confused by the work "missing". Nodes with "missing data" should not be thought of as nodes that are "missing" - they are still present in the output trees. It's simply that the STATE is missing (unknown).

All of which is to say that I find it useful to show an example of how to leave out "missing data" samples, so I've left the second example in, but commented that this gives "topologically connected nodes". Feel free to rephrase or delete as you see fit.